frameworks/kafka/src/main/java/com/mesosphere/sdk/kafka/scheduler/KafkaService.java

```diff
@@ -20,22 +19,31 @@
  */
 public class KafkaService extends DefaultService {
     protected static final Logger LOGGER = LoggerFactory.getLogger(KafkaService.class);
-
+    final KafkaZKClient kafkaZKClient;
 
     public KafkaService(File pathToYamlSpecification) throws Exception {
-        super(pathToYamlSpecification);
+        RawServiceSpec rawServiceSpec = YAMLServiceSpecFactory.generateRawSpecFromYAML(pathToYamlSpecification);
+        DefaultScheduler.Builder schedulerBuilder =
+                DefaultScheduler.newBuilder(YAMLServiceSpecFactory.generateServiceSpec(rawServiceSpec));
+        schedulerBuilder.setPlansFrom(rawServiceSpec);
+
+        kafkaZKClient = new KafkaZKClient(schedulerBuilder.getServiceSpec().getZookeeperConnection(),
+                DcosConstants.SERVICE_ROOT_PATH_PREFIX + schedulerBuilder.getServiceSpec().getName());
+
+        schedulerBuilder.setEndpointProducer("address", new BrokerAddress(kafkaZKClient));
```

Comment:
And PR #647 will add a `native` section which provides direct hosts/IPs to the brokers. See that PR for an example in comments.